pmm-app/src/pmm-check/CheckPanel.test.tsx

```diff
@@ -1,27 +1,17 @@
-import React, { FC } from 'react';
-import { act } from 'react-dom/test-utils';
-import { MemoryRouter, Route } from 'react-router-dom';
+import React from 'react';
 import { mount, ReactWrapper } from 'enzyme';
-import { Spinner } from '@grafana/ui';
-import { Table } from 'pmm-check/components';
-import { ButtonWithSpinner } from 'shared/components/Form';
-import { CheckPanel, CheckPanelProps, CheckPanelState } from './CheckPanel';
+import { CheckPanelRouter } from './CheckPanel';
 import { CheckService } from './Check.service';
 
-import { activeCheckStub } from './__mocks__/stubs';
-
 jest.mock('shared/components/helpers/notification-manager');
 
 jest.mock('./Check.service');
+
 const originalConsoleError = jest.fn();
 
-const CheckPanelRouter: FC<CheckPanelProps> = (props) => (
-  <MemoryRouter>
-    <Route>
-      <CheckPanel {...props} />
-    </Route>
-  </MemoryRouter>
-);
+const dataQa = (label: string) => `[data-qa="${label}"]`;
+
+const runAllPromises = () => new Promise(setImmediate);
```

Comment:
Looks hacky. We need a comment here.

Reply:
I think this solution it's quite used, I'll leave a comment anyway